frontend/src/app/+catalog/+behaviours/system/consolidateKeywords/consolidate-keywords.plugin.ts

```diff
@@ -0,0 +1,155 @@
+import { map } from "rxjs/operators";
+import { MatDialog } from "@angular/material/dialog";
+import { DocEventsService } from "../../../../services/event/doc-events.service";
+import { FormMenuService, MenuId } from "../../../../+form/form-menu.service";
+import { ConfigService } from "../../../../services/config/config.service";
+import { FormUtils } from "../../../../+form/form.utils";
+import { FormStateService } from "../../../../+form/form-state.service";
+import { DocumentService } from "../../../../services/document/document.service";
+import { PluginService } from "../../../../services/plugin/plugin.service";
+import { Plugin } from "../../plugin";
+import { Observable } from "rxjs";
+import { ConsolidateDialogComponent } from "./consolidate-dialog/consolidate-dialog.component";
+
+@Injectable({ providedIn: "root" })
+export class ConsolidateKeywordsPlugin extends Plugin {
+  id = "plugin.consolidate.keywords";
+  name = "Schlagworte konsolidieren";
+  description = "Schlagworte konsolidieren"; // TODO
+  defaultActive = true;
+  forAddress = false; // TODO
+  private readonly isPrivileged: boolean;
+
+  constructor(
+    private docEvents: DocEventsService,
+    private docEventsService: DocEventsService,
+    private documentTreeQuery: TreeQuery,
+    private formMenuService: FormMenuService,
+    private formStateService: FormStateService,
+    private documentService: DocumentService,
+    configService: ConfigService,
+    private dialog: MatDialog,
+  ) {
+    super();
+    inject(PluginService).registerPlugin(this);
+
+    let role = configService.$userInfo.getValue().role;
+    this.isPrivileged =
+      role === "ige-super-admin" || role === "cat-admin" || role === "md-admin";
+  }
+
+  formMenuId: MenuId = "dataset";
+
+  registerForm() {
+    super.registerForm();
+
+    // only add menu item in form if user is privileged and not for address
+    if (this.isPrivileged && !this.forAddress) {
+      const onEvent = this.docEvents
+        .onEvent("OPEN_CONSOLIDATE_KEYWORDS_DIALOG")
+        .subscribe(async (event) => {
+          const handled = await FormUtils.handleDirtyForm(
+            this.formStateService,
+            this.documentService,
+            this.dialog,
+            false,
+          );
+          if (!handled) {
+            return;
+          }
+
+          this.openConsolidateKeywordsDialog(event.data.id).subscribe(
+            async (confirmed) => {
+              if (confirmed) {
+                const handled = await FormUtils.handleDirtyForm(
+                  this.formStateService,
+                  this.documentService,
+                  this.dialog,
+                  this.forAddress,
+                );
+                if (handled)
+                  console.log("consolidate keywords for docId", event.data.id);
+              }
+            },
+          );
+        });
+
+      const onDocLoad = this.documentTreeQuery.openedDocument$.subscribe(
```

Comment:
Das sollte nicht bei jedem Öffnen eines Datensatzes passieren. Menü sollte nur aktualisiert werden, wenn es sich ändert. Ordner müssen auch ausgeschlossen werden, dort gibt es einen Fehler.

Reply:
Die Ordner habe ich nun ausgeschlossen, aber eine Lösung für den anderen Punkt zu finden, wirft andere Probleme auf.
Die anderen Menüpunkte im Dreipunktemenü **Berechtigungen anzeigen** und **Verantwortlichkeit verändern** werden auch bei jedem Laden eines Dokuments erst entfernt und neu hinzugefügt.
Dadurch rutscht der Schlagworte konsolidieren Menüpunkt in der Reihenfolge hoch, wenn man von einem Dokument zu einem anderen klickt.
Kannst du dir das nochmal anschauen ?